### ISSUE TITLE
i2p "Lookup Peers" errors out

### ISSUE THREAD

**Issue by crogonint** [BiglySoftware/BiglyBT#3051]
Java 1.8.0_381 (64 bit)
  Oracle Corporation
c:\program files\java\jre-1.8

SWT v4942r22, win32, zoom=100, dpi=96
Windows 7 v6.1, amd64 (64 bit)
B3.4.0.1_B23/4 az3

I noticed that I've got a couple of little minion looking guys in my list, so I clicked through in to the Privacy Tab, then I click on the Lookup Peers button and it says:

Lookup Status: Error java.lang.Exception: Failed to create server destination, Random is shut down - do you have a static ref?

...I have no idea what to do with that. I tried to look the error up. Nothing. WELL.. hundreds of different java errors. Not this one.

**Comment by parg** [comment]
Sounds like an issue with the I2P plugin, check View->Log Views->I2P Helper Plugin to see if it has any clues

Also check that you are running verison 2.3.0.0.1 of the I2P Helper Plugin

**Comment by crogonint** [comment]
The Settings > Plugins > I2P Helper page says that it is using Version: 2.3.0.1.1 ...Is that the issue?

Well.. I just realized that I'm spoiled with Linux "message types", I had to read that whole log, old school style. ..Nothing jumps out at me. It says there's a router time issue, I'll check momentarily.. but it says it has 0 recognized routers..? Some of the pings are horrifically long. Insanely long, really. I assume that that's just clients too far away to contact.. but shouldn't a ping time out WAY before that? It says that the Tor Proxy DHT client failed a couple of times, but that it succeeded a few times as well. Here's a curated copy of some of the log stuff I either don't know what it is, or looks odd:

[8:27:12] Tor proxy DHT client failed, id=z7odbby2nf32pgic5c7dozai5e46gcuwram4bx252dtwwirxpn6a, cid=300
[8:27:18] Detected router time skew: 49
[8:27:30] Tor proxy DHT client accepted, uid=jfdzhdmyf6qw23xht5nwe3nudr4l445ppbslcowaahqfxmlb4hua, cid=312
[8:28:03] DHT 0: ping:33306,15812,5480,16484,store:2880,2775,105,4306,node:106,64,42,2401,value:115870,44333,29434,30988,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:99207,alien:0,0,0,0,0,0,0/ping:14497,9129,5368,11262,store:852,668,184,1822,node:17027,6490,10469,13567,value:43916,34694,9219,18089,stats:0,0,0,0,data:21,21,0,286,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:1
[8:28:03] DHT 1: ping:28584,12108,4458,6446,store:0,0,0,2554,node:7254,2822,2343,998,value:0,0,0,15809,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:50567,alien:0,0,0,0,0,0,0/ping:10436,3748,6688,4470,store:559,541,18,1582,node:24565,7706,16752,8654,value:0,0,0,9970,stats:0,0,0,0,data:0,0,0,83,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:21
[8:29:03] DHT 0: ping:33320,15812,5480,16484,store:2880,2775,105,4306,node:106,64,42,2401,value:115930,44333,29434,30988,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:99207,alien:0,0,0,0,0,0,0/ping:14499,9129,5370,11262,store:852,668,184,1822,node:17027,6490,10469,13567,value:43916,34694,9219,18089,stats:0,0,0,0,data:21,21,0,286,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:1
[8:29:03] DHT 1: ping:28598,12108,4458,6446,store:0,0,0,2554,node:7254,2822,2343,998,value:0,0,0,15809,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:50567,alien:0,0,0,0,0,0,0/ping:10442,3748,6694,4470,store:559,541,18,1582,node:24565,7706,16752,8654,value:0,0,0,9970,stats:0,0,0,0,data:0,0,0,83,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:21
[8:29:03] Known routers=0, lease-sets=0, caps=NU,ff=false
[8:29:18] Detected router time skew: 49

[8:32:03] Known routers=0, lease-sets=0, caps=NU,ff=false
[8:32:03] Tunnels: exploratory=3, client=0, participating=0
[8:32:19] Detected router time skew: 49
[8:33:03] DHT 0: ping:33376,15812,5480,16484,store:2880,2775,105,4306,node:106,64,42,2401,value:116320,44333,29434,30988,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:99207,alien:0,0,0,0,0,0,0/ping:14507,9129,5378,11262,store:852,668,184,1822,node:17027,6490,10469,13567,value:43916,34694,9219,18089,stats:0,0,0,0,data:21,21,0,286,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:1
[8:33:03] DHT 1: ping:28646,12108,4458,6446,store:0,0,0,2554,node:7271,2822,2343,998,value:0,0,0,15809,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:50567,alien:0,0,0,0,0,0,0/ping:10466,3748,6718,4470,store:559,541,18,1582,node:24565,7706,16752,8654,value:0,0,0,9970,stats:0,0,0,0,data:0,0,0,83,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:21
[8:33:03] Known routers=0, lease-sets=0, caps=NU,ff=false
[8:33:03] Tunnels: exploratory=3, client=0, participating=0

[8:33:21] Tor proxy DHT client failed, id=w6rlgrepctmoy3dcc6nqzytrrfsuhjjr25kkd64rd53bm64qwr5a, cid=316

[8:36:12] Rebuilt global IP bloom filter, size=1001, entries=0, max hits=0
[8:36:13] Rebuilt global IP bloom filter, size=2001, entries=0, max hits=0
[8:36:15] Rebuilt global IP bloom filter, size=2001, entries=82, max hits=4
[8:36:19] Rebuilt global IP bloom filter, size=2001, entries=0, max hits=0
[8:36:19] Rebuilt global IP bloom filter, size=1001, entries=0, max hits=0
[8:36:19] Rebuilt global IP bloom filter, size=1001, entries=55, max hits=3
[8:36:19] Detected router time skew: 49

[8:50:10] Tor proxy DHT client failed, id=jdpodsmhnnf2s3bmw6nlvjds6zouovsfi5ig6xvwqtjsgp2ylqla, cid=320

[8:51:08] Tunnels: exploratory=3, client=0, participating=0
[8:51:12] Rebuilt global IP bloom filter, size=1001, entries=0, max hits=0
[8:51:13] Rebuilt global IP bloom filter, size=2001, entries=0, max hits=0
[8:51:15] Rebuilt global IP bloom filter, size=2001, entries=82, max hits=4
[8:51:19] Rebuilt global IP bloom filter, size=2001, entries=0, max hits=0
[8:51:19] Rebuilt global IP bloom filter, size=1001, entries=0, max hits=0
[8:51:19] Rebuilt global IP bloom filter, size=1001, entries=55, max hits=3
[8:51:29] Detected router time skew: 49

[9:00:36] Republish of cached mappings starts
[9:00:36] Republish of cached mappings completed in 16: values = 0, keys = 0, ops = 0

[9:01:31] Detected router time skew: 49

**Comment by parg** [comment]
For some reason the I2P Router isn't working - hard to say why to be honest. If you're interested in I2P then try running a standalone router and configure BiglyBT to use that.

**Comment by crogonint** [comment]
Is there a setup or diagnostic process I can go through to get the internal one running?

**Comment by parg** [comment]
Well you could restart BiglyBT and look at the Log View from the start, the stuff you posted above had been truncated.

**Comment by crogonint** [comment]
Oooh! Ok!

This is the log immediately after a restart:

[18:14:38] Internal port=17654, external=23154, socks=0, http proxy=0
[18:14:39] Waiting for internal router startup on 127.0.0.1:17654
[18:14:47] Router startup complete: version=2.3.0, elapsed=7486
[18:14:47] [Mix/AZ]: Initializing DHT...
[18:14:59] CPU throttle: factor=2+50/s -> 1ms

This came one minute later:

[18:15:38] Detected router time skew: 0
[18:16:09] [Mix/AZ]: Waiting for socket manager startup
[18:16:09] [Mix/AZ]: Socket manager startup complete - elapsed=81692
[18:16:10] [Mix/AZ]: MyDest: 2rQyr6P07XrSCKu0MUZ44fi--JisUgHjsDoiw~sOSZ3atDKvo~TtetIIq7QxRnjh-L74mKxSAeOwOiLD-w5Jndq0Mq-j9O160girtDFGeOH4vviYrFIB47A6IsP7Dkmd2rQyr6P07XrSCKu0MUZ44fi--JisUgHjsDoiw~sOSZ3atDKvo~TtetIIq7QxRnjh-L74mKxSAeOwOiLD-w5Jndq0Mq-j9O160girtDFGeOH4vviYrFIB47A6IsP7Dkmd2rQyr6P07XrSCKu0MUZ44fi--JisUgHjsDoiw~sOSZ3atDKvo~TtetIIq7QxRnjh-L74mKxSAeOwOiLD-w5Jndq0Mq-j9O160girtDFGeOH4vviYrFIB47A6IsP7Dkmd2rQyr6P07XrSCKu0MUZ44fi--JisUgHjsDoiw~sOSZ3atDKvo~TtetIIq7QxRnjh-L74mKxSAeOwOiLD-w5JnS2GeylAI0tfwwabOKMmjb0uVL1yLXx1R7MnOenQ4p0pBQAEAAcAAA==.i2p
[18:16:10] [Mix/AZ]:         icbhmcbrt35o3bi7xrcspi3gy5nbjlwq6j276esm3v77dhi2vcsa.b32.i2p:59465, existing=true
[18:16:10] [Mix/AZ]: MyNID:  icbhmcgz26c3dr2ekifk5v45fudet63q, existing=true
[18:16:19] [Pure/AZ]: Initializing DHT...
[18:16:23] Tor proxy DHT failed, id=null, cid=1
[18:16:25] [Mix/AZ]+: Initializing DHT...
[18:16:29] Tor proxy DHT failed, id=null, cid=2
[18:16:34] Tor proxy DHT failed, id=null, cid=3
[18:16:38] Detected router time skew: 0
[18:16:40] Tor proxy DHT failed, id=null, cid=4
[18:16:40] [Mix/AZ]+: Waiting for socket manager startup
[18:16:40] [Mix/AZ]+: Socket manager startup complete - elapsed=14870
[18:16:40] [Mix/AZ]+: MyDest+: ZPgn7sjSxpbL6dckJc1uaQ8yTUoYmebIi82XgMv5J5lk-CfuyNLGlsvp1yQlzW5pDzJNShiZ5siLzZeAy~knmWT4J-7I0saWy-nXJCXNbmkPMk1KGJnmyIvNl4DL-SeZZPgn7sjSxpbL6dckJc1uaQ8yTUoYmebIi82XgMv5J5lk-CfuyNLGlsvp1yQlzW5pDzJNShiZ5siLzZeAy~knmWT4J-7I0saWy-nXJCXNbmkPMk1KGJnmyIvNl4DL-SeZZPgn7sjSxpbL6dckJc1uaQ8yTUoYmebIi82XgMv5J5lk-CfuyNLGlsvp1yQlzW5pDzJNShiZ5siLzZeAy~knmWT4J-7I0saWy-nXJCXNbmkPMk1KGJnmyIvNl4DL-SeZZPgn7sjSxpbL6dckJc1uaQ8yTUoYmebIi82XgMv5J5lk-CfuyNLGlsvp1yQlzW5pDzJNShiZ5siLzZeAy~knmUqmUIvipGYSZWN4I2y36ayqP2NHsKoL84aPh6vrtmYnBQAEAAcAAA==.i2p
[18:16:40] [Mix/AZ]+:         xkqnagpnex4uxxdzqql7d4hsgx6ofgwmulbxaynssbfvbzic7k2q.b32.i2p, existing=true
[18:16:40] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Monitoring 'Metadata download for Compiled D&D books.torrent': decentralised
[18:16:40] 3AA35F083EACA49EFBB531FB4451B6214262AE0A.torrent: Monitoring 'Metadata download for D&D.torrent': decentralised
[18:16:40] DHT 0: ping:80,50,8,4,store:0,0,0,0,node:31,15,8,1,value:0,0,0,4,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:11,alien:0,0,0,0,0,0,0/ping:1,0,0,1,store:0,0,0,0,node:0,0,0,0,value:0,0,0,1,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:23
[18:16:40] DHT 1: DHT is inactive
[18:16:40] Known routers=1127, lease-sets=0, caps=LU,ff=false
[18:16:40] Tunnels: exploratory=8, client=10, participating=0
[18:16:40] Throttle: msg_delay=244, tunnel_lag=0, tunnel_stat=Rejecting tunnels: Starting up
[18:16:40] Rates: send=6.5 kB/s, recv=6.3 kB/s, mult=2; Limits: send=100.0 kB/s, recv=100.0 kB/s
[18:16:45] [Pure/AZ]: Waiting for socket manager startup
[18:16:45] [Pure/AZ]: Socket manager startup complete - elapsed=26225
[18:16:45] [Pure/AZ]: MyDest: mhZlyeN9O7ZocprsG7aaHU8i5HdKJRduHMN8kHg1nAaaFmXJ4307tmhymuwbtpodTyLkd0olF24cw3yQeDWcBpoWZcnjfTu2aHKa7Bu2mh1PIuR3SiUXbhzDfJB4NZwGmhZlyeN9O7ZocprsG7aaHU8i5HdKJRduHMN8kHg1nAaaFmXJ4307tmhymuwbtpodTyLkd0olF24cw3yQeDWcBpoWZcnjfTu2aHKa7Bu2mh1PIuR3SiUXbhzDfJB4NZwGmhZlyeN9O7ZocprsG7aaHU8i5HdKJRduHMN8kHg1nAaaFmXJ4307tmhymuwbtpodTyLkd0olF24cw3yQeDWcBpoWZcnjfTu2aHKa7Bu2mh1PIuR3SiUXbhzDfJB4NZwGmhZlyeN9O7ZocprsG7aaHU8i5HdKJRduHMN8kHg1nAaaFmXJ4307tmhymuwbtpodTyLkd0olF24cw3yQeDWcBiUs6s8emYA1UDc3OUsdHoiatBRrK7H5Ja4fC2kFnl7PBQAEAAcAAA==.i2p
[18:16:45] [Pure/AZ]:         rvt57kpbdgnuiqbpyfauixengsb6yzl6d2ppba4f2rcccsn5cseq.b32.i2p:64640, existing=true
[18:16:45] [Pure/AZ]: MyNID:  rvt57ki5th64yxh55tbujipvysrxnfst, existing=true
[18:16:55] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Registering download as Downloading
[18:16:55] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Registration of '' skipped as metadata download
[18:16:55] 3AA35F083EACA49EFBB531FB4451B6214262AE0A.torrent: Registering download as Downloading
[18:16:55] 3AA35F083EACA49EFBB531FB4451B6214262AE0A.torrent: Registration of '' skipped as metadata download
[18:17:00] [Pure/AZ]+: Initializing DHT...
[18:17:01] Tor proxy DHT initialised, id=uooncohynbkvw, cid=5
[18:17:02] [Pure/AZ]+: Waiting for socket manager startup
[18:17:02] [Pure/AZ]+: Socket manager startup complete - elapsed=1526
[18:17:02] [Pure/AZ]+: MyDest+: oXRzTLnGU2ynA3OO5Jk6gMfjEYA2oxhL8R~hFxUM9uShdHNMucZTbKcDc47kmTqAx-MRgDajGEvxH-EXFQz25KF0c0y5xlNspwNzjuSZOoDH4xGANqMYS~Ef4RcVDPbkoXRzTLnGU2ynA3OO5Jk6gMfjEYA2oxhL8R~hFxUM9uShdHNMucZTbKcDc47kmTqAx-MRgDajGEvxH-EXFQz25KF0c0y5xlNspwNzjuSZOoDH4xGANqMYS~Ef4RcVDPbkoXRzTLnGU2ynA3OO5Jk6gMfjEYA2oxhL8R~hFxUM9uShdHNMucZTbKcDc47kmTqAx-MRgDajGEvxH-EXFQz25KF0c0y5xlNspwNzjuSZOoDH4xGANqMYS~Ef4RcVDPbkoXRzTLnGU2ynA3OO5Jk6gMfjEYA2oxhL8R~hFxUM9uShdHNMucZTbKcDc47kmTqAx-MRgDajGEvxH-EXFQz25LGrkt2ex48Hh8vmH2BsHFYKBVO-p8YzqA0A7YhrY54ABQAEAAcAAA==.i2p
[18:17:02] [Pure/AZ]+:         swfjhgoxhqzxeif7qdeycdh737ubybec2hzq3fghjarfctl2bboa.b32.i2p, existing=true

...and this is what got generated over the next few minutes:
[18:17:36] 3AA35F083EACA49EFBB531FB4451B6214262AE0A.torrent: Get of '[Mix/AZ]' completed (elapsed=00:41), peers=0 (0/0/0), seeds=0, leechers=0
[18:17:38] Detected router time skew: 0
[18:17:40] DHT 0: ping:141,91,47,10,store:0,0,0,1,node:31,15,16,4,value:58,26,23,8,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:46,alien:0,0,0,0,0,0,0/ping:15,14,0,7,store:0,0,0,0,node:0,0,0,10,value:29,16,12,6,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:52
[18:17:40] DHT 1: ping:97,68,26,9,store:0,0,0,0,node:35,20,15,0,value:0,0,0,2,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:31,alien:0,0,0,0,0,0,0/ping:0,0,0,12,store:0,0,0,0,node:0,0,0,7,value:0,0,0,1,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:46
[18:17:40] Known routers=1127, lease-sets=0, caps=LU,ff=false
[18:17:40] Tunnels: exploratory=8, client=23, participating=0
[18:17:40] Throttle: msg_delay=244, tunnel_lag=0, tunnel_stat=Rejecting tunnels: Starting up
[18:17:40] Rates: send=12.5 kB/s, recv=21.6 kB/s, mult=2; Limits: send=100.0 kB/s, recv=100.0 kB/s
[18:18:05] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Get of '[Mix/AZ]' completed (elapsed=01:09), peers=0 (0/0/0), seeds=0, leechers=0
[18:18:38] Detected router time skew: 0
[18:18:40] DHT 0: ping:265,161,65,11,store:10,10,0,2,node:148,74,24,4,value:116,52,63,18,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:71,alien:0,0,0,0,0,0,0/ping:88,40,0,11,store:0,0,0,2,node:124,58,8,12,value:40,23,17,10,stats:0,0,0,0,data:1,1,0,1,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:90
[18:18:40] DHT 1: ping:217,144,41,13,store:0,0,0,2,node:140,73,17,0,value:0,0,0,13,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:67,alien:0,0,0,0,0,0,0/ping:7,6,0,15,store:0,0,0,2,node:0,0,0,19,value:0,0,0,3,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:82
[18:18:40] Known routers=1127, lease-sets=0, caps=LU,ff=false
[18:18:40] Tunnels: exploratory=8, client=23, participating=0
[18:18:40] Throttle: msg_delay=139, tunnel_lag=0, tunnel_stat=Rejecting tunnels: Starting up
[18:18:40] Rates: send=20.1 kB/s, recv=24.6 kB/s, mult=2; Limits: send=100.0 kB/s, recv=100.0 kB/s
[18:19:02] Verification of contact 'kjeshanyjzgysgp6rsumkofomrxhiaxhpsyeg4jankzni4sf6erq.b32.i2p:12706' failed for store operation
[18:19:34] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Get of '[Mix/AZ]' completed (elapsed=00:24), peers=0 (0/0/0), seeds=0, leechers=0
[18:19:38] Detected router time skew: 0
[18:19:40] DHT 0: ping:302,198,81,30,store:10,10,0,5,node:180,95,39,5,value:165,74,87,28,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:139,alien:0,0,0,0,0,0,0/ping:143,124,3,24,store:0,0,0,5,node:171,96,17,26,value:50,27,18,15,stats:0,0,0,0,data:1,1,0,1,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:105
[18:19:40] DHT 1: ping:285,203,68,14,store:0,0,0,3,node:200,102,49,0,value:0,0,0,24,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:103,alien:0,0,0,0,0,0,0/ping:105,60,3,18,store:0,0,0,5,node:198,114,7,22,value:0,0,0,11,stats:0,0,0,0,data:0,0,0,6,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:124
[18:19:40] Known routers=1127, lease-sets=0, caps=LU,ff=false
[18:19:40] Tunnels: exploratory=8, client=23, participating=0
[18:19:40] Throttle: msg_delay=94, tunnel_lag=0, tunnel_stat=Rejecting tunnels: Starting up
[18:19:40] Rates: send=27.5 kB/s, recv=29.7 kB/s, mult=2; Limits: send=100.0 kB/s, recv=100.0 kB/s
[18:20:38] Detected router time skew: 0
[18:20:42] DHT 0: ping:322,212,101,46,store:10,10,0,5,node:214,107,68,7,value:165,74,90,47,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:201,alien:0,0,0,0,0,0,0/ping:159,142,7,33,store:0,0,0,7,node:188,102,28,34,value:54,27,22,21,stats:0,0,0,0,data:1,1,0,1,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:109
[18:20:42] DHT 1: ping:311,223,76,15,store:0,0,0,5,node:228,119,74,1,value:0,0,0,33,stats:0,0,0,0,data:0,0,0,0,kb:0,0,0,0,incoming:139,alien:0,0,0,0,0,0,0/ping:121,103,10,19,store:0,0,0,6,node:230,137,21,27,value:0,0,0,21,stats:0,0,0,0,data:0,0,0,12,kb:0,0,0,0,incoming:0,alien:0,0,0,0,0,0,0:126
[18:20:42] Known routers=1129, lease-sets=0, caps=LU,ff=false
[18:20:42] Tunnels: exploratory=9, client=27, participating=0
[18:20:42] Throttle: msg_delay=94, tunnel_lag=0, tunnel_stat=Rejecting tunnels: Starting up
[18:20:42] Rates: send=17.9 kB/s, recv=19.8 kB/s, mult=2; Limits: send=100.0 kB/s, recv=100.0 kB/s
[18:20:53] 44814B2D70E43A337815016F0E7E016D700FFC3A.torrent: Get of '[Mix/AZ]' completed (elapsed=00:11), peers=0 (0/0/0), seeds=0, leechers=0

The only things I erased was two lines about registering torrents, because above it says not to list torrents. The rest of the log generated since then looks like the same junk I posted above.. but I can totally do a pastebin or whatever if you want to see the whole thing. Not a problem, just let me know what you need from me. :)

**Comment by parg** [comment]
Actually that log is showing I2P working OK (known routers=1129, tunnels created, send/recv rate not zero...)

Try doing a peer-lookup now, see if it responds any better

**Comment by crogonint** [comment]
HEY! It did a thing! Now what?
![i2p](https://github.com/BiglySoftware/BiglyBT/assets/23371790/49b9a5d1-daf8-4cd8-af4c-7cf0eef55200)
On my sources tab it says I2P 0/26/0 and it lists 26 leechers. It didn't start downloading anything, so I assume these are all people I was already connected to? ..or do I need to do something else to connect to these people?

**Comment by crogonint** [comment]
Actually.. it looks like all of the torrents say I2P 0/8/0 or whatever. Am I doing anything by running the internal I2P manually? Now, I don't understand why the little minion guy told me to go to the privacy tab, if it's already doing its thing by itself.

..besides the fact that my I2P was broken the first time I checked it.

**Comment by crogonint** [comment]
Oh, I'm in a DMZ (behind two routers) Someone has to try to hack me on purpose to reach me, so I'm not at all concerned about privacy. I have all of the networks and everything enabled in settings.. that I know how to do. Thank you for explaining why the guy is there though. :)

More things I'd like to know:
1) How come when I limit leechers with pieces I want to say 60kbps, it keeps downloading at 400kbps?

2) How do I connect to and download from these guys?
![BiglyBT Peers](https://github.com/BiglySoftware/BiglyBT/assets/23371790/457b57d1-d54e-424f-82f2-f294b28e9ac9)
(This is why I was trying to figure out if the minion guy could help me.)

**Comment by crogonint** [comment]
![peers](https://github.com/BiglySoftware/BiglyBT/assets/23371790/b4a663a9-aff8-4200-b81c-4e0e85173767)
I limited this bottom guy (and my public thing) to 60kbps up, but it's still going to 400-500kbps.
Guys like this complete their download in minutes, then I ever get the pieces I need.

**Comment by crogonint** [comment]
Some of those 0 seeds and few peers have actual peers with pieces that are needed, but they grab the pieces they need, then disconnect. I' having to manually throttle them down to stupid slow speeds to get all of the pieces they have that I need. Of course, even then, if they catch on that I've throttled them down, they'll just disconnect anyway.. sometimes. I was assuming that the rest are people I can't connect to.. but perhaps they're phantoms /residuals that the trackers are hanging on to, as you said.

I figured out that the Boost overrides the Upload speed limits. I'm not sure if it does anything for download.

Didn't we used to be able to request personal chats with people? I can't figure out how to bump or nudge somebody to get their attention to try to share pieces and etc.
